tests/ImageSharp.Tests/Processing/Transforms/TransformTests.cs

```diff
@@ -142,13 +168,27 @@ public void Transform_WithSampler<TPixel>(TestImageProvider<TPixel> provider, st
                 Matrix3x2 rotate = Matrix3x2Extensions.CreateRotationDegrees(50);
                 Matrix3x2 scale = Matrix3x2Extensions.CreateScale(new SizeF(.5F, .5F));
                 var translate = Matrix3x2.CreateTranslation(75, 0);
-
-
+                
                 image.Mutate(i => i.Transform(rotate * scale * translate, sampler));
                 image.DebugSave(provider, resamplerName);
             }
         }
 
+        private Matrix3x2 MakeManuallyCenteredMatrix<TPixel>(float angleDeg, float s, Image<TPixel> image)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Matrix3x2 rotate = Matrix3x2Extensions.CreateRotationDegrees(angleDeg);
+            Vector2 toCenter = 0.5f * new Vector2(image.Width, image.Height);
+            var translate = Matrix3x2.CreateTranslation(-toCenter);
+            var translateBack = Matrix3x2.CreateTranslation(toCenter);
+            var scale = Matrix3x2.CreateScale(s);
+
+            Matrix3x2 m = translate * rotate * scale * translateBack;
```

Comment:
I don't think this is correct. The last translation back will be incorrect due to pre-scaling. 

In order to calculate the correct matrix you would need to also alter the last translate back to take into consideration the new dimensions.

`var translateBack = Matrix3x2.CreateTranslation(toCenter * s); `

Reply:
@JimBobSquarePants this transform does not care with target image dimensions. An example:
If we have a 10x10 shape in the center of an 50x50 image and we want to scale it down by the factor of `0.5` keeping it in the center of the original 50x50 canvas, we move it to the origo (`(-25,-25)`), scale it, than move it back with a`(+25,+25)` vector.

In my tests the cropping is done manually by defining the new canvas "after" the transform.